skbuild/setuptools_wrap.py

```diff
@@ -444,10 +466,17 @@ def setup(*args, **kw):  # noqa: C901
     try:
         cmkr = cmaker.CMaker()
         if not skip_cmake:
-            env = cmkr.configure(cmake_args,
-                                 cmake_source_dir=cmake_source_dir,
-                                 cmake_install_dir=skbuild_kw['cmake_install_dir'],
-                                 languages=cmake_languages)
+            # skip the configure step for a cached build
+            env = cmkr.get_cached_env()
+            if env is None or cmake_args != _load_cmake_args():
```

Comment:
I suggest to abort mentioning that a different version of cmake is being used, or attempt re-configure and let CMake report the message

Reply:
The current approach is to re-configure, so I'll do that.